Program.cs

```diff
@@ -1,143 +1,83 @@
+        var path = "C:\\Users\\niels\\Downloads\\espn-bears.m4a";
+        var uploadedFileUrl = await UploadFileAsync(path);
+        var transcript = await GetTranscriptAsync(uploadedFileUrl);
+        Console.WriteLine("Transcript:\n" + transcript);
+    }
 
-        TranscriptFetcher transcriptFetcher = new TranscriptFetcher();
+    private static async Task<string> UploadFileAsync(string path)
+    {
+        await using var fileStream = File.OpenRead(path);
+        using var fileContent = new StreamContent(fileStream);
+        fileContent.Headers.ContentType = new MediaTypeHeaderValue("application/octet-stream");
 
-        try
-        {
-            dynamic transcript = await transcriptFetcher.GetTranscriptAsync(apiToken, uploadUrl);
+        using var response = await HttpClient.PostAsync("https://api.assemblyai.com/v2/upload", fileContent);
 
-            Console.WriteLine("Transcript:\n" + transcript.text);
-        }
-        catch (Exception ex)
+        if (response.IsSuccessStatusCode == false)
+            throw new Exception($"Error: {response.StatusCode} - {response.ReasonPhrase}");
+
+        var jsonDoc = await response.Content.ReadFromJsonAsync<JsonDocument>();
+        return jsonDoc.RootElement.GetProperty("upload_url").GetString();
+    }
+
+    private static async Task<string> GetTranscriptAsync(string audioUrl)
+    {
+        var data = new {audio_url = audioUrl};
+        var content = new StringContent(JsonSerializer.Serialize(data), Encoding.UTF8, "application/json");
+
+        using var response = await HttpClient.PostAsync("https://api.assemblyai.com/v2/transcript", content);
+        var responseJson = await response.Content.ReadFromJsonAsync<JsonDocument>();
+
+        var transcriptId = responseJson.RootElement.GetProperty("id").GetString();
+        var pollingEndpoint = $"https://api.assemblyai.com/v2/transcript/{transcriptId}";
+
+        while (true)
         {
-            Console.WriteLine($"Error: {ex.Message}");
+            var pollingResponse = await HttpClient.GetAsync(pollingEndpoint);
+            var pollingJsonDocument = await pollingResponse.Content.ReadFromJsonAsync<JsonDocument>();
+            var pollingJsonObject = pollingJsonDocument.RootElement;
+
+            var status = pollingJsonObject.GetProperty("status").GetString();
+            switch (status)
+            {
+                case "processing": 
+                    await Task.Delay(TimeSpan.FromSeconds(3));
+                    break;
+                case "completed": 
+                    return pollingJsonObject.GetProperty("text").GetString();
+                case "error":
+                    var error = pollingJsonObject.GetProperty("error").GetString();
+                    throw new Exception($"Transcription failed: {error}");
+                default:
+                    throw new Exception("This code should not be reachable.");
+            }
```

Comment:
Handling all cases explicitly, rather than assuming the status is processing when it is note completed or error.